File: my_cpp_pkg/src/number_system/add_two_ints_server.cpp
#include <rclcpp/rclcpp.hpp>
#include <example_interfaces/srv/add_two_ints.hpp>


using namespace example_interfaces::srv;

class AddTwoIntsServer : public rclcpp::Node{
   private:
    rclcpp::Service<AddTwoInts>::SharedPtr server_;

    void callbackAddTwoInts(const AddTwoInts::Request::SharedPtr req, AddTwoInts::Response::SharedPtr res) {
        res->sum = req->a + req->b;
        RCLCPP_INFO(this->get_logger(), "%ld + %ld = %ld", req->a, req->b, res->sum);
    }

   public:
    AddTwoIntsServer() : Node("add_two_ints_server") {
        server_ = this->create_service<AddTwoInts>("add_two_ints", std::bind(&AddTwoIntsServer::callbackAddTwoInts, this, std::placeholders::_1, std::placeholders::_2));

        RCLCPP_INFO(this->get_logger(), "add_two_ints_server Started...");
    }
};


int main(int argc, char** argv) {
    rclcpp::init(argc, argv);

    auto node = std::make_shared<AddTwoIntsServer>();
    rclcpp::spin(node);

    rclcpp::shutdown();
    
    return 0;
}

File: my_py_pkg/my_py_pkg/add_two_ints_server.py
import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts


class AddTwoIntsServer(Node):
    def __init__(self):
        super().__init__('add_two_ints_server')

        self.create_service(AddTwoInts, 'add_two_ints', self.callback_add_two_ints)
        self.get_logger().info('add_two_ints_server started')


    def callback_add_two_ints(self, req: AddTwoInts.Request, res: AddTwoInts.Response):
        res.sum = req.a + req.b
        self.get_logger().info(f"{req.a} + {req.b} = {res.sum}")
        return res




def main(args=None):
    rclpy.init(args=args)

    node = AddTwoIntsServer()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == '__main__':
    main()